File: 백준/Bronze/2292. 벌집/벌집.py
def find_min_steps(N):
    if N == 1:
        return 1  
    
    layer = 1 
    max_number = 1  
    
    while max_number < N:
        max_number += 6 * layer  
        layer += 1 
    
    return layer

N = int(input())
print(find_min_steps(N))

File: 백준/Bronze/2292. 벌집/벌집.kt
fun findMinSteps(N: Int): Int {
    if (N == 1) return 1  

    var layer = 1 
    var maxNumber = 1  

    while (maxNumber < N) {
        maxNumber += 6 * layer  
        layer++ 
    }

    return layer
}

fun main() {
    val N = readLine()!!.toInt()
    println(findMinSteps(N))
}
